libraries/api-documenter/src/yaml/OfficeYamlDocumenter.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as fsx from 'fs-extra';
+import * as path from 'path';
+import yaml = require('js-yaml');
+
+import { DocItemSet } from '../utils/DocItemSet';
+import { IYamlTocItem } from './IYamlTocFile';
+import { IYamlItem } from './IYamlApiFile';
+import { YamlDocumenter } from './YamlDocumenter';
+
+interface ISnippetsFile {
+  /**
+   * The keys are API names like "Excel.Range.clear".
+   * The values are TypeScript source code excerpts.
+   */
+  [apiName: string]: string[];
+}
+
+/**
+ * Extends YamlDocumenter with some custom logic that is specific to Office Add-ins.
+ */
+export class OfficeYamlDocumenter extends YamlDocumenter {
```

Comment:
>OfficeYamlDocumenter [](start = 13, length = 20)

Ideally this should belong to a separate tool that extends YamlDocumenter. api-documenter should be a tools for general-purpose and does snot include al the customizations.

Reply:
If you can follow up, make sure this one gets moved out of this repo. 

---
In reply to: [146029384](https://github.com/Microsoft/web-build-tools/pull/405#discussion_r146029384) [](ancestors = 146029384)

Reply:
I agree.  The simplest long term solution is probably for Office to set up their own separate TypeScript project that extends the YamlDocumenter and ApiDocumenterCommandLine classes in the same way we are doing here.  That project could live in the [office-js](https://github.com/OfficeDev/office-js) repo.  The code changes would be minimal, but maintaining a cross-repo relationship requires more coordination.

This PR is just an experiment with @Zlatkovsky to try an idea.  If it doesn't work out, we'll revert the changes.  Or if it ends up meeting their needs, then that would justify putting more polish into the design.

The **api-documenter** tool itself is meant to be a code sample showing different ways that API JSON can be processed, so I don't mind a little cruft in there.

---
In reply to: [146029495](https://github.com/Microsoft/web-build-tools/pull/405#discussion_r146029495) [](ancestors = 146029495,146029384)